crd/nodeinfo/client.go

```diff
@@ -0,0 +1,161 @@
+
+func init() {
+	_ = scheme.AddToScheme(Scheme)
+	_ = v1alpha1.AddToScheme(Scheme)
+}
+
+// Installer provides methods to manage the lifecycle of the NodeInfo resource definition.
+type Installer struct {
+	cli typedv1.CustomResourceDefinitionInterface
+}
+
+func NewInstaller(c *rest.Config) (*Installer, error) {
+	cli, err := crd.NewCRDClientFromConfig(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init crd client")
+	}
+	return &Installer{
+		cli: cli,
+	}, nil
+}
+
+func (i *Installer) create(ctx context.Context, res *v1.CustomResourceDefinition) (*v1.CustomResourceDefinition, error) {
+	res, err := i.cli.Create(ctx, res, metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create nodeinfo crd")
+	}
+	return res, nil
+}
+
+// Install installs the embedded NodeInfo CRD definition in the cluster.
+func (i *Installer) Install(ctx context.Context) (*v1.CustomResourceDefinition, error) {
+	nodeinfo, err := GetNodeInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get embedded nodeinfo crd")
+	}
+	return i.create(ctx, nodeinfo)
+}
+
+// InstallOrUpdate installs the embedded NodeInfo CRD definition in the cluster or updates it if present.
+func (i *Installer) InstallOrUpdate(ctx context.Context) (*v1.CustomResourceDefinition, error) {
+	nodeinfo, err := GetNodeInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get embedded nodeinfo crd")
+	}
+	current, err := i.create(ctx, nodeinfo)
+	if !apierrors.IsAlreadyExists(err) {
+		return current, err
+	}
+	if current == nil {
+		current, err = i.cli.Get(ctx, nodeinfo.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get existing nodeinfo crd")
+		}
+	}
+	if !reflect.DeepEqual(nodeinfo.Spec.Versions, current.Spec.Versions) {
+		nodeinfo.SetResourceVersion(current.GetResourceVersion())
+		previous := *current
+		current, err = i.cli.Update(ctx, nodeinfo, metav1.UpdateOptions{})
+		if err != nil {
+			return &previous, errors.Wrap(err, "failed to update existing nodeinfo crd")
+		}
+	}
+	return current, nil
+}
+
+// Client provides methods to interact with instances of the NodeInfo custom resource.
+type Client struct {
+	cli client.Client
+}
+
+// NewClient creates a new NodeInfo client around the passed ctrlcli.Client.
+func NewClient(cli client.Client) *Client {
+	return &Client{
+		cli: cli,
+	}
+}
+
+// Get returns the NodeInfo identified by the NamespacedName.
+func (c *Client) Get(ctx context.Context, key types.NamespacedName) (*v1alpha1.NodeInfo, error) {
+	nodeInfo := &v1alpha1.NodeInfo{}
+	err := c.cli.Get(ctx, key, nodeInfo)
+	return nodeInfo, errors.Wrapf(err, "failed to get nodeinfo %v", key)
+}
+
+// PatchSpec performs a server-side patch of the passed NodeInfoSpec to the NodeInfo specified by the NamespacedName.
+func (c *Client) PatchSpec(ctx context.Context, key types.NamespacedName, spec *v1alpha1.NodeInfoSpec, fieldManager string) (*v1alpha1.NodeInfo, error) {
```

Comment:
not necessary as we will not be patching the spec

Reply:
removed